sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -246,4 +247,45 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     }
   }
 
+  test("datetime function current_date") {
+    val d0 = DateTimeUtils.millisToDays(System.currentTimeMillis())
+    val cd = CurrentDate().eval(EmptyRow).asInstanceOf[Int]
+    val d1 = DateTimeUtils.millisToDays(System.currentTimeMillis())
+    assert(d0 <= cd && cd <= d1 && d1 - d0 <= 1)
+  }
+
+  test("datetime function current_timestamp") {
+    val ct = DateTimeUtils.toJavaTimestamp(CurrentTimestamp().eval(EmptyRow).asInstanceOf[Long])
+    val t1 = System.currentTimeMillis()
+    assert(math.abs(t1 - ct.getTime) < 5000)
+  }
+
+  test("function to_date") {
+    checkResult(ToDate(Literal("2015-07-22")), ToDate(Literal("2015-07-22 23:59:59")))
+    checkResult(
+      ToDate(Literal("2015-07-22")), ToDate(Literal(Timestamp.valueOf("2015-07-22 23:59:59"))))
+    checkResult(ToDate(Literal("2015-07-22")), Date.valueOf("2015-07-22"))
+  }
+
+  test("function trunc") {
+    checkResult(
```

Comment:
Seems this will not trigger the codegen projections, you probably need to use the `checkEvaluation` instead.